newsplease/helper_classes/parse_crawler.py

```diff
@@ -40,51 +44,53 @@ def pass_to_pipeline_if_article(
         :return NewscrawlerItem: NewscrawlerItem to pass to the pipeline
         """
         if self.helper.heuristics.is_article(response, original_url):
-            return self.pass_to_pipeline(
-                response, source_domain, rss_title=None)
+            return self.pass_to_pipeline(response, source_domain, rss_title=rss_title)
 
     def pass_to_pipeline(
-            self,
-            response,
-            source_domain,
-            rss_title=None
+        self, response: Response, source_domain: str, rss_title: Optional[str] = None
     ):
-        timestamp = time.strftime('%Y-%m-%d %H:%M:%S',
-                                  time.gmtime(time.time()))
+        timestamp = time.strftime("%Y-%m-%d %H:%M:%S", time.gmtime(time.time()))
 
-        relative_local_path = self.helper.savepath_parser \
-            .get_savepath(response.url)
+        relative_local_path = self.helper.savepath_parser.get_savepath(response.url)
 
         # Instantiate the crawler item class defined in the configuration
         article = self.helper.crawler_item_class()
-        article['local_path'] = self.helper.savepath_parser \
-            .get_formatted_relative_path(relative_local_path)
-        article['filename'] = self.helper.savepath_parser.get_filename(article['local_path'])
-        article['abs_local_path'] = self.helper.savepath_parser \
-            .get_abs_path(relative_local_path)
-        article['modified_date'] = timestamp
-        article['download_date'] = timestamp
-        article['source_domain'] = source_domain.encode("utf-8")
-        article['url'] = response.url
-        extracted_title = response.selector.xpath('//title/text()').extract_first()
-        article['html_title'] = extracted_title.encode("utf-8") if extracted_title is not None else ''
+        article["local_path"] = self.helper.savepath_parser.get_formatted_relative_path(
+            relative_local_path
+        )
+        article["filename"] = self.helper.savepath_parser.get_filename(
+            article["local_path"]
+        )
+        article["abs_local_path"] = self.helper.savepath_parser.get_abs_path(
+            relative_local_path
+        )
+        article["modified_date"] = timestamp
+        article["download_date"] = timestamp
+        article["source_domain"] = source_domain.encode("utf-8")
+        article["url"] = response.url
+        extracted_title = response.selector.xpath("//title/text()").extract_first()
+        if extracted_title is None:
+            extracted_title = ""
+        article["html_title"] = extracted_title.encode("utf-8")
```

Comment:
Here was the issue. Whatever the result of the xpath().extract_first(), we handle a string that will be encoded at the end